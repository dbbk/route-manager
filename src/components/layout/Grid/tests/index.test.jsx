import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Grid, GridElement } from '../';

describe('<Grid />', () => {
  test('with initial props', () => {
    const wrapper = shallow(<Grid />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with 1 column', () => {
    const wrapper = shallow(
      <Grid>
        <GridElement width={1 / 1}>Item 1</GridElement>
        <GridElement width={1 / 1}>Item 2</GridElement>
        <GridElement width={1 / 1}>Item 3</GridElement>
        <GridElement width={1 / 1}>Item 4</GridElement>
        <GridElement width={1 / 1}>Item 5</GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with 2 columns', () => {
    const wrapper = shallow(
      <Grid>
        <GridElement width={1 / 2}>Item 1</GridElement>
        <GridElement width={1 / 2}>Item 2</GridElement>
        <GridElement width={1 / 2}>Item 3</GridElement>
        <GridElement width={1 / 2}>Item 4</GridElement>
        <GridElement width={1 / 2}>Item 5</GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with 3 columns', () => {
    const wrapper = shallow(
      <Grid>
        <GridElement width={1 / 3}>Item 1</GridElement>
        <GridElement width={1 / 3}>Item 2</GridElement>
        <GridElement width={1 / 3}>Item 3</GridElement>
        <GridElement width={1 / 3}>Item 4</GridElement>
        <GridElement width={1 / 3}>Item 5</GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with 3 columns and span', () => {
    const wrapper = shallow(
      <Grid>
        <GridElement width={2 / 3}>Item 1</GridElement>
        <GridElement width={1 / 3}>Item 2</GridElement>
        <GridElement width={1 / 3}>Item 3</GridElement>
        <GridElement width={1 / 3}>Item 4</GridElement>
        <GridElement width={1 / 3}>Item 5</GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with 4 columns', () => {
    const wrapper = shallow(
      <Grid>
        <GridElement width={1 / 4}>Item 1</GridElement>
        <GridElement width={1 / 4}>Item 2</GridElement>
        <GridElement width={1 / 4}>Item 3</GridElement>
        <GridElement width={1 / 4}>Item 4</GridElement>
        <GridElement width={1 / 4}>Item 5</GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with 5 columns', () => {
    const wrapper = shallow(
      <Grid>
        <GridElement width={1 / 5}>Item 1</GridElement>
        <GridElement width={1 / 5}>Item 2</GridElement>
        <GridElement width={1 / 5}>Item 3</GridElement>
        <GridElement width={1 / 5}>Item 4</GridElement>
        <GridElement width={1 / 5}>Item 5</GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with custom Grid element', () => {
    const wrapper = shallow(
      <Grid element="article">
        <GridElement width={1 / 2}>Item 1</GridElement>
        <GridElement width={1 / 2}>Item 2</GridElement>
        <GridElement width={1 / 2}>Item 3</GridElement>
        <GridElement width={1 / 2}>Item 4</GridElement>
        <GridElement width={1 / 2}>Item 5</GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with custom Grid and GridElement elements', () => {
    const wrapper = shallow(
      <Grid element="article">
        <GridElement element="section" width={1 / 2}>
          Item 1
        </GridElement>
        <GridElement element="section" width={1 / 2}>
          Item 2
        </GridElement>
        <GridElement element="section" width={1 / 2}>
          Item 3
        </GridElement>
        <GridElement element="section" width={1 / 2}>
          Item 4
        </GridElement>
        <GridElement element="section" width={1 / 2}>
          Item 5
        </GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('with responsive columns', () => {
    const wrapper = shallow(
      <Grid>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          First Element
        </GridElement>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          Second Element
        </GridElement>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          Third Element
        </GridElement>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          Fourth Element
        </GridElement>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          Fifth Element
        </GridElement>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          Sixth Element
        </GridElement>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          Seventh Element
        </GridElement>
        <GridElement
          width={{
            sm: 1 / 2,
            md: 1 / 3,
            lg: 1 / 4,
            xlg: 1 / 5,
          }}
        >
          Eighth Element
        </GridElement>
      </Grid>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
