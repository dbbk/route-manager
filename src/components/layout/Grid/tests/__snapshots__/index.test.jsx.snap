// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> with 1 column 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
>
  <Component
    width={1}
  >
    Item 1
  </Component>
  <Component
    width={1}
  >
    Item 2
  </Component>
  <Component
    width={1}
  >
    Item 3
  </Component>
  <Component
    width={1}
  >
    Item 4
  </Component>
  <Component
    width={1}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with 2 columns 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
>
  <Component
    width={0.5}
  >
    Item 1
  </Component>
  <Component
    width={0.5}
  >
    Item 2
  </Component>
  <Component
    width={0.5}
  >
    Item 3
  </Component>
  <Component
    width={0.5}
  >
    Item 4
  </Component>
  <Component
    width={0.5}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with 3 columns 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
>
  <Component
    width={0.3333333333333333}
  >
    Item 1
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 2
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 3
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 4
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with 3 columns and span 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
>
  <Component
    width={0.6666666666666666}
  >
    Item 1
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 2
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 3
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 4
  </Component>
  <Component
    width={0.3333333333333333}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with 4 columns 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
>
  <Component
    width={0.25}
  >
    Item 1
  </Component>
  <Component
    width={0.25}
  >
    Item 2
  </Component>
  <Component
    width={0.25}
  >
    Item 3
  </Component>
  <Component
    width={0.25}
  >
    Item 4
  </Component>
  <Component
    width={0.25}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with 5 columns 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
>
  <Component
    width={0.2}
  >
    Item 1
  </Component>
  <Component
    width={0.2}
  >
    Item 2
  </Component>
  <Component
    width={0.2}
  >
    Item 3
  </Component>
  <Component
    width={0.2}
  >
    Item 4
  </Component>
  <Component
    width={0.2}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with custom Grid and GridElement elements 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
  element="article"
>
  <Component
    element="section"
    width={0.5}
  >
    Item 1
  </Component>
  <Component
    element="section"
    width={0.5}
  >
    Item 2
  </Component>
  <Component
    element="section"
    width={0.5}
  >
    Item 3
  </Component>
  <Component
    element="section"
    width={0.5}
  >
    Item 4
  </Component>
  <Component
    element="section"
    width={0.5}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with custom Grid element 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
  element="article"
>
  <Component
    width={0.5}
  >
    Item 1
  </Component>
  <Component
    width={0.5}
  >
    Item 2
  </Component>
  <Component
    width={0.5}
  >
    Item 3
  </Component>
  <Component
    width={0.5}
  >
    Item 4
  </Component>
  <Component
    width={0.5}
  >
    Item 5
  </Component>
</Primitive>
`;

exports[`<Grid /> with initial props 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
/>
`;

exports[`<Grid /> with responsive columns 1`] = `
.c0 {
  margin: -15px 0;
}

@media (min-width:568px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    margin: -15px;
  }
}

<Primitive
  className="c0"
>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    First Element
  </Component>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    Second Element
  </Component>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    Third Element
  </Component>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    Fourth Element
  </Component>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    Fifth Element
  </Component>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    Sixth Element
  </Component>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    Seventh Element
  </Component>
  <Component
    width={
      Object {
        "lg": 0.25,
        "md": 0.3333333333333333,
        "sm": 0.5,
        "xlg": 0.2,
      }
    }
  >
    Eighth Element
  </Component>
</Primitive>
`;
